src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticWebAssetsIntegrationTest.cs

```diff
@@ -0,0 +1,269 @@
+        }
+
+        public Task InitializeAsync()
+        {
+            return PackageTestProjects.PackAsync(Output);
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        private string GetExpectedManifest()
+        {
+            var restorePath = LocalNugetPackagesCacheTempPath;
+            var projects = new[]
+            {
+                Path.Combine(restorePath, "packagelibrarytransitivedependency", "1.0.0", "buildTransitive", "..", "razorContent") + Path.DirectorySeparatorChar,
+                Path.Combine(restorePath, "packagelibrarydirectdependency", "1.0.0", "build", "..", "razorContent") + Path.DirectorySeparatorChar
+            };
+
+            return $@"<StaticWebAssets Version=""1.0"">
+  <ContentRoot BasePath=""_content/PackageLibraryTransitiveDependency"" Path=""{projects[0]}"" />
+  <ContentRoot BasePath=""_content/PackageLibraryDirectDependency"" Path=""{projects[1]}"" />
+</StaticWebAssets>";
+        }
+    }
+
+    public class PackageTestProjectsFixture
+    {
+        private bool _packed;
+
+        internal async Task PackAsync(ITestOutputHelper output)
+        {
+            if (_packed)
+            {
+                return;
+            }
+
+            var projectsToPack = GetProjectsToPack();
+
+            foreach (var project in projectsToPack)
+            {
+                output.WriteLine(project);
+            }
+
+            foreach (var project in projectsToPack)
+            {
+                var psi = new ProcessStartInfo
+                {
+                    FileName = DotNetMuxer.MuxerPathOrDefault(),
+#if DEBUG
+                    Arguments = "msbuild /t:Restore;Pack /p:Configuration=Debug",
+#else
+                    Arguments = "msbuild /t:Restore;Pack /p:Configuration=Release",
+#endif
+                    WorkingDirectory = project,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true
+                };
+                var tcs = new TaskCompletionSource<Process>();
+                var process = Process.Start(psi);
+                process.Exited += (s, a) =>
+                {
+                    tcs.SetResult(process);
+                };
+                process.EnableRaisingEvents = true;
+
+                await Task.WhenAny(Task.Delay(TimeSpan.FromMinutes(2)), tcs.Task);
+
+                // Wait for 10 seconds or fail. If we take longer
+                // it likely means we are adding unexpected dependencies.
+                Assert.True(process.HasExited, "Process did not finish running.");
+                output.WriteLine(process.StandardOutput.ReadToEnd());
+                Assert.Equal(0, process.ExitCode);
+            }
+
+            _packed = true;
+        }
+
+        public static string[] GetProjectsToPack()
+        {
+            return typeof(PackageTestProjectsFixture).Assembly.GetCustomAttributes<AssemblyMetadataAttribute>()
```

Comment:
Can we just use paths here?

Reply:
Id rather not, that makes tests brittle when we move things around. We've been using this technique all over the place with good results, so I rather keep it this way.